src/neo/Consensus/ConsensusContext.cs

```diff
@@ -88,8 +88,8 @@ public Block CreateBlock()
             for (int i = 0, j = 0; i < Validators.Length && j < M; i++)
             {
                 if (CommitPayloads[i]?.ConsensusMessage.ViewNumber != ViewNumber) continue;
-                sc.AddSignature(contract, Validators[i], CommitPayloads[i].GetDeserializedMessage<Commit>().Signature);
-                j++;
+                if (sc.AddSignature(contract, Validators[i], CommitPayloads[i].GetDeserializedMessage<Commit>().Signature))
```

Comment:
Under what kind of attack will the addition fail?

Reply:
No attack, just an error.

Reply:
I notice this error with a wrong blockchain imported, and a valid CN open (Related to https://github.com/neo-project/neo/issues/1414 )

Reply:
Imported? 

Strange, Shargon, did it really solve the problem?
Because the import does not pass through here, it goes to OnPersist and it is another flow, right?

Reply:
This doesn't solve the problem of wrong import, only I notice that this variable is increased wrong if signature was not added

Reply:
I checked this before and I found it strange as well, because some commit could be sent with wrong signature.

But I think this will not be enough in such case, I believe we would need to clean Header and exit the Return Block, right?